sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -202,4 +204,230 @@ class FileBasedDataSourceSuite extends QueryTest with SharedSQLContext with Befo
       }
     }
   }
+
+  // Unsupported data types of csv, json, orc, and parquet are as follows;
+  //  csv -> R/W: Interval, Null, Array, Map, Struct
+  //  json -> W: Interval
+  //  orc -> W: Interval, Null
+  //  parquet -> R/W: Interval, Null
+  test("SPARK-24204 error handling for unsupported Array/Map/Struct types - csv") {
+    withTempDir { dir =>
+      val csvDir = new File(dir, "csv").getCanonicalPath
```

Comment:
This is duplicated with `CSVSuite`.

Reply:
This tests are moved from `CSVSuite`: https://github.com/apache/spark/pull/21389/files#diff-219ac8201e443435499123f96e94d29fL743

Do I misunderstand your comment?

Reply:
No, you are right. I misunderstood.

Reply:
Thanks!